tests/unit/test_cephfs_interfaces.py

```diff
@@ -0,0 +1,253 @@
+        self.assertEqual(
+            event.share_info,
+            CephFSShareInfo(fsid=FSID, name=NAME, path=PATH, monitor_hosts=MONITOR_HOSTS),
+        )
+        self.assertEqual(event.auth_info, CephFSAuthInfo(username=USERNAME, key=KEY))
+
+    @patch.object(CephFSClientCharm, "_on_umount_share")
+    def test_on_umount_share(self, _on_umount_share) -> None:
+        """Assert that correct hook is called when CephFS share is ready to be unmounted."""
+        # Disconnect the related unit to trigger `relation-departed`.
+        self.harness.remove_relation_unit(self.integration_id, "application/0")
+
+        # Simulate the auth secret
+        auth_id = self.harness.add_model_secret(
+            "application", CephFSAuthInfo(username=USERNAME, key=KEY).dict()
+        )
+        self.harness.grant_secret(auth_id, self.harness.charm.app)
+
+        # Simulate CephFS share endpoint being present.
+        self.harness.update_relation_data(
+            self.integration_id,
+            "application",
+            {
+                "share_info": SHARE_INFO,
+                "auth_id": auth_id,
+            },
+        )
+
+        # Assert the correct event handler is invoked and event parameters are passed.
+        _on_umount_share.assert_called_once()
+        event = _on_umount_share.call_args[0][0]
+        self.assertEqual(
+            event.share_info,
+            CephFSShareInfo(fsid=FSID, name=NAME, path=PATH, monitor_hosts=MONITOR_HOSTS),
+        )
+        self.assertEqual(event.auth_info, CephFSAuthInfo(username=USERNAME, key=KEY))
+
+    def test_request_share(self) -> None:
+        """Assert that name is in integration databag."""
+        # Evaluate if all values are set.
+        self.harness.charm.requirer.request_share(self.integration_id, "/data")
+        self.assertDictEqual(
+            self.harness.get_relation_data(self.integration_id, "cephfs-client"),
+            {"name": "/data"},
+        )
+
+    def tearDown(self) -> None:
+        self.harness.cleanup()
+
+
+class TestCephFSServerCharm(unittest.TestCase):
+    """Test cephfs_interfaces from the provider side."""
+
+    def setUp(self):
+        self.harness = Harness(CephFSServerCharm, meta=CEPHFS_SERVER_METADATA)
+        self.integration_id = self.harness.add_relation(CEPHFS_INTEGRATION_NAME, "application")
+        self.harness.add_relation_unit(self.integration_id, "application/0")
+        self.harness.set_leader(True)
+        self.harness.begin_with_initial_hooks()
+
+    @patch.object(CephFSServerCharm, "_on_share_requested")
+    def test_on_share_requested(self, _on_share_requested) -> None:
+        """Assert that correct hook is called when a new CephFS share is requested."""
+        # Simulate the request of a new CephFS share.
+        self.harness.update_relation_data(
+            self.integration_id,
+            "application",
+            {
+                "name": (name := "/data"),
```

Comment:
[nit]: it's my personal preference to only use the walrus operator `:=` when the declared variable is scoped to an if-else or match-case branch. There's nothing technically wrong with using the walrus operator here, but I find that explicitly declaring `name` with `name = "/data"` above `self.harness.update_relation_data(...)` is better for the readability of this test case. Explicit is better than implicit, even if explicit adds a couple more lines to the code.

You can read up more about the controversy of the walrus operator here; it's a hot button topic for some Python programmers :sweat_smile:: https://www.codeconquest.com/blog/python-walrus-operator-uses-and-controversy/#htoc-observations-against-the-walrus-operator-in-python